### PR TITLE
Global-Edge-Rewiring

### PR DESCRIPTION
**Pull request by coryglover**
This adds the standard Watts-Strogatz method.

### CODE REVIEW

**Comment by coryglover** [comment]
@jkbren I updated this to be "Global Edge Rewiring" per our lunch conversation.

**Comment by sdmccabe** [comment]
Can you remove the `.DS_Store` files and the untitled notebook from the commit?

**Comment by sdmccabe** [comment]
I'll do a separate PR for the gitignore.

File: netrw/rewire/global_rewiring.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            new_edges = {}
+
+        # Give every edge opportunity to change
+        if timesteps == -1:
+            timesteps = len(list(G.edges()))
+
+        # Rewire at each timestep
+        for t in range(timesteps):
+            # Decide whether to rewire
+            if p < random.random():
```
Comment (jkbren) [inline netrw/rewire/global_rewiring.py:57]: `if p > random.random():`

**Comment by jkbren** [comment]
Cory and I just talked and walked through how to change this to be `step_rewire` and `full_rewire`. Tested it on several networks, both `verbose=True` and not, and it looks amazing 🤌

He's gonna re-push these edits and think through how to do the same for the other techniques implemented.

**Review by jkbren — APPROVED** [review]
Great! This is a good example if folks need one moving forward.

**Pull request merged**